js/README.md

```diff
@@ -100,55 +100,163 @@ Subdirectories with additional utilities
 
-1. Define the type of block
+1. Create a new class inside the file. All demonstrations extend from `BaseBlock`, however blocks can extend each other also.
 
-`var uniquenameBlock = new ProtoBlock('uniquename');`
+e.g. `class uniqueNameBlock extends AnotherBlock{}`
 
-e.g., `var pitchNumberBlock = new ProtoBlock('pitchnumber');`
+* As part of MVC structure the argument earlier passed to `new ProtoBlock()` is now passed to `super()` call inside constructor of the defined class.
 
+`super(uniquename);`
+
+<!-- `var uniquenameBlock = new ProtoBlock('uniquename');` -->
+
+<!-- e.g., `var pitchNumberBlock = new ProtoBlock('pitchnumber');` -->
+<!-- 
 This creates a new instance of the class protoBlock, which is used to
-create instances of the block.
+create instances of the block. -->
 
 2. Assign a palette to the block
 
-`uniquenameBlock.palette = palettes.dict['yourpalettename'];`
+<!-- `uniquenameBlock.palette = palettes.dict['yourpalettename'];` -->
+
+<!-- e.g., `pitchNumberBlock.palette = palettes.dict['pitch'];` -->
+e.g. `this.setPalette('yourPalleteName);`
+
+* Your class definition should look similar to this:
 
-e.g., `pitchNumberBlock.palette = palettes.dict['pitch'];`
+```
+  class NamedDoBlock extends FlowBlock{
+    constructor() {
+      super(nameddo);
+      this.setPalette('action');
+    }
+  }
+```
+Note: After the new update there is no requirement for a `beginnerMode` check as `BaseBlock` automatically performs that check. 
 
 The palette can be any of the palettes listed in `turtledef.js`. The
 color of the block is defined by the palette used.
 
-3. To add block to the protoblock dictionary
+3. Add a call to `new myNewBlock.setup()` in the previously defined `setup` function.
+
+e.g. `function setupActionBlocks() {`
+
+  `new CalcBlock().setup();`
+
+`}`
+
+* For arg blocks, define a function `arg` inside the block class definition. There are 4 arguments currently passed to this function viz. `(logo, turtle, blk, receivedArg)`.
+
+e.g. 
+
+```
+  class CalcBlock extends LeftBlock {
+    constructor() {
+        super('calc');
+        this.setPalette('action');
+    }
+
+    arg(logo, turtle, blk, receivedArg) {
+    }
+}
+```
+
+* For flow bocks define a function `flow` on the block. The same 4 arguments are passed to the flow function currently: `(logo, turtle, blk, receivedArg)`.
+
+e.g 
+```
+class ListenBlock extends FlowBlock {
+    constructor() {
+        super('listen');
+        this.setPalette('action');
+    }
+
+    flow(args, logo, turtle, blk, receivedArg) {
+    }
+}
+```
+
+Note: Trailing arguments can be neglected in both functions, if not needed.
+
+<!-- To add block to the protoblock dictionary
 
 `blocks.protoBlockDict['uniquename'] = uniquenameBlock;`
+ -->
+<!-- e.g., `blocks.protoBlockDict['pitchnumber'] = pitchNumberBlock;` -->
+
+4. Write the logic for the block in either of the two functions, `arg()` or `flow()`.
+
+Note: Change of syntax after MVC structure:
```

Comment:
Again. not sure you need to explain the old way. Just the new way.

Reply:
Changing this too.